rest-api-spec/src/main/resources/rest-api-spec/test/cat.shards/10_basic.yml

```diff
@@ -69,6 +69,7 @@
              search.concurrent_query_current  .+   \n
              search.concurrent_query_time     .+   \n
              search.concurrent_query_total    .+   \n
+             search.avg_query_concurrency     .+   \n
```

Comment:
@jed326 @sohami looking into this metric, the name seems to be very vague, may be `search.concurrent_avg_slices_per_query` is a better one?

Reply:
@reta agreed that the current name is vague. I like the suggestion to maintain the same prefix `"search.concurrent_*"` for all the concurrent search metrics too but I'm hesitant about mentioning "slices" since I think that's an internal concept that a user probably doesn't understand. How about something like `search.concurrent_avg_threads_per_query`? Although the same thread can pick up multiple slices too so that might be confusing as well...

Reply:
:+1:  May be: 
```
search.concurrent_avg_parallelism_per_query
```
?

Reply:
Unfortunately this metric is only gathering the slice count so it's not really a measure of parallelism either. How about 

```
search.concurrent_avg_thread_executions_per_query
```
?

We already introduced the term "thread_executions" back in https://github.com/opensearch-project/OpenSearch/pull/7673 as "The number of threads that have been scheduled to work on the task" so it seems reasonable to re-use the same terminology here.

Reply:
Totally agree, it was not really a measure of parallelism

Reply:
:+1 to `search.concurrent_avg_thread_executions_per_query`, @sohami you are OK with it?

Reply:
How about using the `slice` keyword for this metric. fe: `search.concurrent_avg_slice_count`. We have the [cluster setting to control the slice count ](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/SearchService.java#L263)per query, so users will be familiar with that. The metric is actually reflecting that.

Reply:
That makes sense to me, I forgot we were already exposing the "slice" concept via that setting so this should be fine. I also think it's fine to exclude "per_query" since the rest of the metrics are query level so it seems clear to me that this is per query.